apps/clamav.sh

```diff
@@ -0,0 +1,185 @@
+    if is_app_installed files_antivirus
+    then
+        occ_command_no_check app:remove files_antivirus
+    fi
+    # Show successful uninstall if applicable
+    removal_popup
+else
+    print_text_in_color "$ICyan" "Installing ClamAV..."
+fi
+
+# Install needed tools
+apt update -q4 & spinner_loading
+apt install clamav clamav-freshclam clamav-daemon -y
+
+# stop freshclam and update the database
+check_command systemctl stop clamav-freshclam
+check_command freshclam
+check_command systemctl start clamav-freshclam
+
+# Edit ClamAV settings to fit the installation
+sed -i "s|^MaxDirectoryRecursion.*|MaxDirectoryRecursion 30|" /etc/clamav/clamd.conf
+sed -i "s|^MaxFileSize.*|MaxFileSize 100M|" /etc/clamav/clamd.conf
+sed -i "s|^PCREMaxFileSize.*|PCREMaxFileSize 100M|" /etc/clamav/clamd.conf
+sed -i "s|^StreamMaxLength.*|StreamMaxLength 100M|" /etc/clamav/clamd.conf
+
+# Start ClamAV
+check_command systemctl restart clamav-freshclam
+check_command systemctl restart clamav-daemon
+
+# Make the service more reliable
+check_command cp /lib/systemd/system/clamav-daemon.service /etc/systemd/system/clamav-daemon.service
+sed -i '/\[Service\]/a Restart=always' /etc/systemd/system/clamav-daemon.service
+sed -i '/\[Service\]/a RestartSec=3' /etc/systemd/system/clamav-daemon.service
+check_command systemctl daemon-reload
+check_command systemctl restart clamav-daemon
+
+# Install Nextcloud app
+install_and_enable_app files_antivirus
+
+# Configure Nextcloud app
+occ_command config:app:set files_antivirus av_mode --value="socket"
+occ_command config:app:set files_antivirus av_socket --value="/var/run/clamav/clamd.ctl"
+occ_command config:app:set files_antivirus av_stream_max_length --value="104857600"
+occ_command config:app:set files_antivirus av_max_file_size --value="-1"
+occ_command config:app:set files_antivirus av_infected_action --value="only_log"
+
+# Inform the user
+msg_box "ClamAV was succesfully installed.
+
+Your Nextcloud should be safer now."
+
+# Ask for full-scan
+if ! yesno_box_yes "Do you want to setup a weekly full-scan of all files?
+It will run on sundays starting at 10:00 for max 12h. 
+You will be notified what the result was."
+then
+    exit
+fi
+
+choice=$(whiptail --title "$TITLE" --nocancel --menu \
+"Choose what shall happen with infected files.
+Infected files will always get reported to you no matter which option you choose.
+$MENU_GUIDE\n\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+"Only log" "" \
+"Copy to a folder" "" \
+"Move to a folder" "" \
+"Remove" "" 3>&1 1>&2 2>&3)
+
+case "$choice" in
+    "Only log")
+        ARGUMENT=""
+    ;;
+    "Copy to a folder")
+        ARGUMENT="--move="
+    ;;
+    "Move to a folder")
+        ARGUMENT="--copy="
+    ;;
+    "Remove")
+        ARGUMENT="--remove=yes"
+    ;;
+    "")
+        exit 1
+    ;;
+    *)
+    ;;
+esac
+
+if [ "$choice" = "Copy to a folder" ] || [ "$choice" = "Move to a folder" ]
+then
+    AV_PATH="/root/.clamav/clamav-fullscan.jail"
```

Comment:
Is this location good?

Reply:
Doesn't clamav have it's on folder?

Is this for logging?

Reply:
I am not sure if clamav has its own folder.
no, this is for storing the infected files.